diff --git a/plugins/imtcp/imtcp.c b/plugins/imtcp/imtcp.c
index abd42d5..d2d8d92 100644
--- a/plugins/imtcp/imtcp.c
+++ b/plugins/imtcp/imtcp.c
@@ -230,6 +230,15 @@ static struct cnfparamdescr modpdescr[] = {{"flowcontrol", eCmdHdlrBinary, 0},
                                            {"streamdriver.crlfile", eCmdHdlrString, 0},
                                            {"streamdriver.keyfile", eCmdHdlrString, 0},
                                            {"streamdriver.certfile", eCmdHdlrString, 0},
+                                           {"StreamDriverMode", eCmdHdlrNonNegInt, 0},
+                                           {"StreamDriverAuthMode", eCmdHdlrString, 0},
+                                           {"StreamDriver", eCmdHdlrString, 0},
+                                           {"StreamDriverCAFile", eCmdHdlrString, 0},
+                                           {"StreamDriverCRLFile", eCmdHdlrString, 0},
+                                           {"StreamDriverKeyFile", eCmdHdlrString, 0},
+                                           {"StreamDriverCertFile", eCmdHdlrString, 0},
+                                           {"StreamDriverPermittedPeers", eCmdHdlrArray, 0},
+                                           {"StreamDriverPermittedPeer", eCmdHdlrArray, 0},
                                            {"permittedpeer", eCmdHdlrArray, 0},
                                            {"keepalive", eCmdHdlrBinary, 0},
                                            {"keepalive.probes", eCmdHdlrNonNegInt, 0},
@@ -270,6 +279,15 @@ static struct cnfparamdescr inppdescr[] = {{"port", eCmdHdlrString, CNFPARAM_REQ
                                            {"streamdriver.crlfile", eCmdHdlrString, 0},
                                            {"streamdriver.keyfile", eCmdHdlrString, 0},
                                            {"streamdriver.certfile", eCmdHdlrString, 0},
+                                           {"StreamDriverMode", eCmdHdlrNonNegInt, 0},
+                                           {"StreamDriverAuthMode", eCmdHdlrString, 0},
+                                           {"StreamDriver", eCmdHdlrString, 0},
+                                           {"StreamDriverCAFile", eCmdHdlrString, 0},
+                                           {"StreamDriverCRLFile", eCmdHdlrString, 0},
+                                           {"StreamDriverKeyFile", eCmdHdlrString, 0},
+                                           {"StreamDriverCertFile", eCmdHdlrString, 0},
+                                           {"StreamDriverPermittedPeers", eCmdHdlrArray, 0},
+                                           {"StreamDriverPermittedPeer", eCmdHdlrArray, 0},
                                            {"permittedpeer", eCmdHdlrArray, 0},
                                            {"gnutlsprioritystring", eCmdHdlrString, 0},
                                            {"keepalive", eCmdHdlrBinary, 0},
@@ -617,7 +635,8 @@ BEGINnewInpInst
             inst->bSPFramingFix = (int)pvals[i].val.d.n;
         } else if (!strcmp(inppblk.descr[i].name, "ruleset")) {
             inst->pszBindRuleset = (uchar *)es_str2cstr(pvals[i].val.d.estr, NULL);
-        } else if (!strcmp(inppblk.descr[i].name, "streamdriver.mode")) {
+        } else if (!strcmp(inppblk.descr[i].name, "streamdriver.mode") ||
+                   !strcmp(inppblk.descr[i].name, "StreamDriverMode")) {
             inst->iStrmDrvrMode = (int)pvals[i].val.d.n;
         } else if (!strcmp(inppblk.descr[i].name, "streamdriver.CheckExtendedKeyPurpose")) {
             inst->iStrmDrvrExtendedCertCheck = (int)pvals[i].val.d.n;
@@ -629,25 +648,33 @@ BEGINnewInpInst
             } else {
                 parser_errmsg("streamdriver.TlsVerifyDepth must be 2 or higher but is %d", (int)pvals[i].val.d.n);
             }
-        } else if (!strcmp(inppblk.descr[i].name, "streamdriver.authmode")) {
+        } else if (!strcmp(inppblk.descr[i].name, "streamdriver.authmode") ||
+                   !strcmp(inppblk.descr[i].name, "StreamDriverAuthMode")) {
             inst->pszStrmDrvrAuthMode = (uchar *)es_str2cstr(pvals[i].val.d.estr, NULL);
         } else if (!strcmp(inppblk.descr[i].name, "streamdriver.permitexpiredcerts")) {
             inst->pszStrmDrvrPermitExpiredCerts = (uchar *)es_str2cstr(pvals[i].val.d.estr, NULL);
-        } else if (!strcmp(inppblk.descr[i].name, "streamdriver.cafile")) {
+        } else if (!strcmp(inppblk.descr[i].name, "streamdriver.cafile") ||
+                   !strcmp(inppblk.descr[i].name, "StreamDriverCAFile")) {
             inst->pszStrmDrvrCAFile = (uchar *)es_str2cstr(pvals[i].val.d.estr, NULL);
-        } else if (!strcmp(inppblk.descr[i].name, "streamdriver.crlfile")) {
+        } else if (!strcmp(inppblk.descr[i].name, "streamdriver.crlfile") ||
+                   !strcmp(inppblk.descr[i].name, "StreamDriverCRLFile")) {
             inst->pszStrmDrvrCRLFile = (uchar *)es_str2cstr(pvals[i].val.d.estr, NULL);
-        } else if (!strcmp(inppblk.descr[i].name, "streamdriver.keyfile")) {
+        } else if (!strcmp(inppblk.descr[i].name, "streamdriver.keyfile") ||
+                   !strcmp(inppblk.descr[i].name, "StreamDriverKeyFile")) {
             inst->pszStrmDrvrKeyFile = (uchar *)es_str2cstr(pvals[i].val.d.estr, NULL);
-        } else if (!strcmp(inppblk.descr[i].name, "streamdriver.certfile")) {
+        } else if (!strcmp(inppblk.descr[i].name, "streamdriver.certfile") ||
+                   !strcmp(inppblk.descr[i].name, "StreamDriverCertFile")) {
             inst->pszStrmDrvrCertFile = (uchar *)es_str2cstr(pvals[i].val.d.estr, NULL);
-        } else if (!strcmp(inppblk.descr[i].name, "streamdriver.name")) {
+        } else if (!strcmp(inppblk.descr[i].name, "streamdriver.name") ||
+                   !strcmp(inppblk.descr[i].name, "StreamDriver")) {
             inst->pszStrmDrvrName = (uchar *)es_str2cstr(pvals[i].val.d.estr, NULL);
         } else if (!strcmp(inppblk.descr[i].name, "starvationprotection.maxreads")) {
             inst->starvationMaxReads = (unsigned)pvals[i].val.d.n;
         } else if (!strcmp(inppblk.descr[i].name, "gnutlsprioritystring")) {
             inst->gnutlsPriorityString = (uchar *)es_str2cstr(pvals[i].val.d.estr, NULL);
-        } else if (!strcmp(inppblk.descr[i].name, "permittedpeer")) {
+        } else if (!strcmp(inppblk.descr[i].name, "permittedpeer") ||
+                   !strcmp(inppblk.descr[i].name, "StreamDriverPermittedPeers") ||
+                   !strcmp(inppblk.descr[i].name, "StreamDriverPermittedPeer")) {
             for (int j = 0; j < pvals[i].val.d.ar->nmemb; ++j) {
                 uchar *const peer = (uchar *)es_str2cstr(pvals[i].val.d.ar->arr[j], NULL);
                 CHKiRet(net.AddPermittedPeer(&inst->pPermPeersRoot, peer));
@@ -822,7 +849,8 @@ BEGINsetModCnf
             loadModConf->gnutlsPriorityString = (uchar *)es_str2cstr(pvals[i].val.d.estr, NULL);
         } else if (!strcmp(modpblk.descr[i].name, "networknamespace")) {
             loadModConf->pszNetworkNamespace = es_str2cstr(pvals[i].val.d.estr, NULL);
-        } else if (!strcmp(modpblk.descr[i].name, "streamdriver.mode")) {
+        } else if (!strcmp(modpblk.descr[i].name, "streamdriver.mode") ||
+                   !strcmp(modpblk.descr[i].name, "StreamDriverMode")) {
             loadModConf->iStrmDrvrMode = (int)pvals[i].val.d.n;
         } else if (!strcmp(modpblk.descr[i].name, "streamdriver.CheckExtendedKeyPurpose")) {
             loadModConf->iStrmDrvrExtendedCertCheck = (int)pvals[i].val.d.n;
@@ -834,21 +862,29 @@ BEGINsetModCnf
             } else {
                 parser_errmsg("streamdriver.TlsVerifyDepth must be 2 or higher but is %d", (int)pvals[i].val.d.n);
             }
-        } else if (!strcmp(modpblk.descr[i].name, "streamdriver.authmode")) {
+        } else if (!strcmp(modpblk.descr[i].name, "streamdriver.authmode") ||
+                   !strcmp(modpblk.descr[i].name, "StreamDriverAuthMode")) {
             loadModConf->pszStrmDrvrAuthMode = (uchar *)es_str2cstr(pvals[i].val.d.estr, NULL);
         } else if (!strcmp(modpblk.descr[i].name, "streamdriver.permitexpiredcerts")) {
             loadModConf->pszStrmDrvrPermitExpiredCerts = (uchar *)es_str2cstr(pvals[i].val.d.estr, NULL);
-        } else if (!strcmp(modpblk.descr[i].name, "streamdriver.cafile")) {
+        } else if (!strcmp(modpblk.descr[i].name, "streamdriver.cafile") ||
+                   !strcmp(modpblk.descr[i].name, "StreamDriverCAFile")) {
             loadModConf->pszStrmDrvrCAFile = (uchar *)es_str2cstr(pvals[i].val.d.estr, NULL);
-        } else if (!strcmp(modpblk.descr[i].name, "streamdriver.crlfile")) {
+        } else if (!strcmp(modpblk.descr[i].name, "streamdriver.crlfile") ||
+                   !strcmp(modpblk.descr[i].name, "StreamDriverCRLFile")) {
             loadModConf->pszStrmDrvrCRLFile = (uchar *)es_str2cstr(pvals[i].val.d.estr, NULL);
-        } else if (!strcmp(modpblk.descr[i].name, "streamdriver.keyfile")) {
+        } else if (!strcmp(modpblk.descr[i].name, "streamdriver.keyfile") ||
+                   !strcmp(modpblk.descr[i].name, "StreamDriverKeyFile")) {
             loadModConf->pszStrmDrvrKeyFile = (uchar *)es_str2cstr(pvals[i].val.d.estr, NULL);
-        } else if (!strcmp(modpblk.descr[i].name, "streamdriver.certfile")) {
+        } else if (!strcmp(modpblk.descr[i].name, "streamdriver.certfile") ||
+                   !strcmp(modpblk.descr[i].name, "StreamDriverCertFile")) {
             loadModConf->pszStrmDrvrCertFile = (uchar *)es_str2cstr(pvals[i].val.d.estr, NULL);
-        } else if (!strcmp(modpblk.descr[i].name, "streamdriver.name")) {
+        } else if (!strcmp(modpblk.descr[i].name, "streamdriver.name") ||
+                   !strcmp(modpblk.descr[i].name, "StreamDriver")) {
             loadModConf->pszStrmDrvrName = (uchar *)es_str2cstr(pvals[i].val.d.estr, NULL);
-        } else if (!strcmp(modpblk.descr[i].name, "permittedpeer")) {
+        } else if (!strcmp(modpblk.descr[i].name, "permittedpeer") ||
+                   !strcmp(modpblk.descr[i].name, "StreamDriverPermittedPeers") ||
+                   !strcmp(modpblk.descr[i].name, "StreamDriverPermittedPeer")) {
             for (int j = 0; j < pvals[i].val.d.ar->nmemb; ++j) {
                 uchar *const peer = (uchar *)es_str2cstr(pvals[i].val.d.ar->arr[j], NULL);
                 CHKiRet(net.AddPermittedPeer(&loadModConf->pPermPeersRoot, peer));
